Add authentication test

mario-galic opened the pull request:
@emma-p @markjamesbutler 

Added authentication test which simply checks if Google Account prompt page
opens up when not authenticated.

Split tests into two categories 'Small' and 'Large': please see [test README](https://github.com/guardian/identity-admin/tree/test-auth/test/README.md)

emma-p commented on build.sbt:
```diff
@@ -25,3 +26,9 @@ play.PlayImport.PlayKeys.playDefaultPort := 8852
 
 addCommandAlias("devrun", "run -Dconfig.resource=dev.conf -Dlogs.home=logs")
+
+javaOptions in Test += "-Dconfig.file=test/conf/application.conf"
```
We could also have this instead: `javaOptions in Test += "-Dconfig.resource=TEST.conf"`
And have `TEST.conf` reference `application.conf`, it would be consistent with the way it works for other ENVs

emma-p commented on test/conf/application.conf:
```diff
@@ -0,0 +1,8 @@
+include file("conf/application.conf")
```
Same as above, would it be possible to have this information in a `conf/TEST.conf` file?

emma-p commented on test/large/AuthSpec.scala:
```diff
@@ -0,0 +1,52 @@
+      Given("the user has not been previously authenticated")
+      assert(!hasSessionCookie)
+
+      When("attempting to search ID database")
```
Very good idea to add this test, maybe it's worth adding one for all the major endpoints (delete for example)

danmassie commented:
I don't think we should be redefining the different types of testing. Sbt provides a mechanism for separating unit and integration tests and also provides useful commands such as testQuick which rely upon the built in test commands and scopes. Requiring the use of command aliases to filter which tests are run prevents using these built in features without aliasing all combinations. I also disagree about in-browser tests, I always want them to be a separate task to running unit or even integration tests. They can't for example be run on a CI server without pointing at a running instance and that running instance should be the same version of the code currently being built.

See http://www.scala-sbt.org/0.12.4/docs/Detailed-Topics/Testing.html for sbt testing options.

mario-galic commented:
@danmassie Per request, I've reverted testing guidelines.

mario-galic commented:
@markjamesbutler I moved acceptance tests into a separate sub-project `functional-tests`. They can be executed with `sbt "project functional-tests" test`. I reverted all changes to the main project.

Works in [TC](http://teamcity.guprod.gnm/viewLog.html?buildId=743640&buildTypeId=Identity_IdentitySanityTests_IdentityAdmin&tab=buildLog)

markjamesbutler commented:
:+1: LGTM
